lib/progressive_hash_diff.rb

```diff
@@ -0,0 +1,30 @@
+class ProgressiveHashDiff
```

Comment:
Hmm, yeah, this makes more sense.

I think I would've gone for `HashDiffBuilder` with a more fluent interface like:

```ruby
HashDiffBuilder.new
  .previous_item(a)
  .current_item(b)
  .diff
```